server/service/base_client.go

```diff
@@ -65,7 +75,49 @@ func (bc *baseClient) url(path, rawQuery string) *url.URL {
 	return &u
 }
 
-func newBaseClient(addr string, insecureSkipVerify bool, rootCA, urlPrefix string) (*baseClient, error) {
+// setServerCapabilities updates the server capabilities based on the response
+// from the server.
+func (bc *baseClient) setServerCapabilities(response *http.Response) {
+	bc.serverCapabilities = map[fleet.Capability]struct{}{}
+	capabilities := response.Header.Get("X-Fleet-Capabilities")
+
+	if capabilities == "" {
```

Comment:
This is not needed, if `capabilities == ""` then the next for loop will never iterate

Reply:
you're right, thanks! removed.